### PR TITLE
ENH JTFS `unpad_frequency` in PyTorch and TensorFlow

### PR DESCRIPTION
**Pull request by lostanlen**
segue of #955 
`jtfs_average_and_format` is now fully ported to all backends

### CODE REVIEW

**Comment by lostanlen** [comment]
TODO 
[*] port "unpad_frequency" to PyTorch and TF backends
[ ] write tensorflow frontend
[ ] test tensorflow differentiability
[ ] write keras frontend
[ ] write sklearn frontend
[ ] write backend.split in all backends
[ ] support out_type='list' in base frontend

**Comment by lostanlen** [comment]
@cyrusvahidi @MuawizChaudhary @changhongw @lylyhan 

may i focus on the last two items (`backend.split` and  `format='time'`) and count on some/all of you to write the frontend + TF differentiability test?

File: kymatio/scattering1d/backend/torch_backend.py [hunk 1]
```diff
@@ -174,13 +174,35 @@ def swap_time_frequency(cls, x):
...
+    @staticmethod
+    def unpad_frequency(x, n1_max, n1_stride):
+        """Unpad the frequency axis after frequency scattering and/or frequency
+        averaging. This is called at the end of `jtfs_average_and_format`
+        unless `out_type='array'` and `format='joint'`.
+        NB. Unpadding is one-sided. See point 3 of pad_frequency docstring.
+        Parameters
+        ----------
+        x : tensor (batch, frequency, time, 1), corresponds to path['coef']
+        n1_max: integer. last first-order wavelet index n1 such that j1 < j2
```
Comment (changhongw) [inline kymatio/scattering1d/backend/torch_backend.py:195]: `n1_max: integer, the last first-order wavelet index (n1) such that j1 < j2`

File: kymatio/scattering1d/backend/torch_backend.py [hunk 2]
```diff
@@ -174,13 +174,35 @@ def swap_time_frequency(cls, x):
...
+        for the scattering path of x. By definition, lower than len(psi1_f).
+        n1_stride: integer frequential subsampling factor associated to the
```
Comment (changhongw) [inline kymatio/scattering1d/backend/torch_backend.py:197]: `n1_stride: integer, related to the frequential subsampling factor associated to the`

`n1_stride` itself is not the frequential subsampling factor

**Review by changhongw — APPROVED** [review]

**Review by cyrusvahidi — APPROVED** [review]

File: kymatio/scattering1d/backend/torch_backend.py (hunk 2 above)
Comment (lostanlen) [reply kymatio/scattering1d/backend/torch_backend.py:197]: > n1_stride itself is not the frequential subsampling factor

i think it is? the `n1` values of `U1` are spaced by `1` whereas the `Y_fr` are spaced by `n1_stride`, so the subsampling factor is `n1_stride`

Comment (changhongw) [reply kymatio/scattering1d/backend/torch_backend.py]: Ah, yes, my bad.

**Pull request merged**